src/inbound.rs

```diff
@@ -137,7 +137,8 @@ impl Inbound {
                 if let Ok((reader, writer, info)) = server.try_connect(&self.shared_state.options) {
                     // replace our reader and writer to correspond with the new socket
                     self.reader = reader;
-                    self.info = info;
+                    self.info = info.clone();
+                    *self.shared_state.info.write() = info;
```

Comment:
Why do we have it in two places now?

Reply:
I was thinking it would be OK to avoid the mutex in Inbound, but that doesn't matter because this is likely to be called so infrequently from Inbound that it doesn't matter. I'll take it out of there to keep it only present in one place.